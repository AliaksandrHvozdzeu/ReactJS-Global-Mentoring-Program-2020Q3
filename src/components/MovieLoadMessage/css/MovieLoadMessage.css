.modal-window-loader {
    font-size: 2rem;
    text-align: center;
}

.modal-window {
    background-color: var(--grey-darker);
    color: var(--white);
    position: fixed;
    left: 50vw;
    top: 50vh;
    max-height: 80vh;
    min-width: 500px;
    width: 40%;
    transform: translate(-50%, -50%);
    border-radius: 4px;
    box-shadow: 0 0 10px 0 #000000b3;
    padding: 4rem 5rem;
    overflow: auto;
    z-index: 2;
}

.modal-window-overlay-message {
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background-color: var(--grey-darker);
    filter: blur(1px) grayscale(50%);
    opacity: 0.8;
    z-index: 2;
}


element.style {
}
header > * {
    position: relative;
    z-index: 2;
}
header {
    height: 50vh;
    background-image: url(/images/header.jpg);
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover;
    position: relative;
    overflow: hidden;
}
* {
    margin: 0;
    padding: 0;
    border: 0;
    box-sizing: border-box;
}
user agent stylesheet
header {
    display: block;
}
body {
    background-color: var(--grey);
    min-height: 100vh;
    position: relative;
    font-family: 'Roboto', sans-serif;
}
html {
    --red: #f65261;
    --red-lighter: #fb6672;
    --red-darker: #e62b3a;
    --grey: #555555;
    --grey-transparent: rgba(85, 85, 85, 0.44);
    --grey-lighter-transparent: #70707070;
    --grey-darker: #232323;
    --white: #ffffff;
    --grey-darker-transparent: #232323d7;
    --black-darker-transparent: rgb(21 20 20 / 93%);
}

