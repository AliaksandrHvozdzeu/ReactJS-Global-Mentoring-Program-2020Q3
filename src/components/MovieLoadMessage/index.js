import MovieLoadMessage from './js/MovieLoadMessage';

export default MovieLoadMessage;
