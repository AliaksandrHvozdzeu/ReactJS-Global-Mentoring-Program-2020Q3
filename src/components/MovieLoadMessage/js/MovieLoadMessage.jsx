import React, { useEffect } from 'react';
import '../css/MovieLoadMessage.css';

const MovieLoadMessage = () => {

  useEffect(() => {
    const container = document.body;
    container.style.overflow = 'hidden';
    return () => {container.style.overflow = 'auto';};
  }, []);

  return (
    <>
      <div className="modal-window-overlay-message"/>
      <div className="modal-window">
        <div className="modal-window-loader">Loading... Please wait!</div>
      </div>
    </>
  );

};

export default MovieLoadMessage;
