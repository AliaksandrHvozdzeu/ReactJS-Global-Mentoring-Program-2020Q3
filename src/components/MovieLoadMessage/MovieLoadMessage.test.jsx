import React from 'react';
import rerender from 'react-test-renderer';
import MovieLoadMessage from './js/MovieLoadMessage';
import { mount } from "enzyme";

describe('when MovieLoadMessage', () => {

  test('then snapshot created', () => {
    const component = rerender.create(<MovieLoadMessage/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('containers with hidden style', () => {
    const container = document.body;
    container.style.overflow = 'hidden';
    expect(container.style.overflow).toEqual('hidden');
  });

  test('MovieLoadMessage useEffect', () => {
    const wrapper = mount(<MovieLoadMessage />);
    expect(wrapper.find('.modal-window-loader').text()).toBe('Loading... Please wait!');
    wrapper.setProps({
      importantProp: 'new value',
    });
    wrapper.update();
    expect(wrapper.find('.modal-window-loader').text()).toBe('Loading... Please wait!');
  });

});
