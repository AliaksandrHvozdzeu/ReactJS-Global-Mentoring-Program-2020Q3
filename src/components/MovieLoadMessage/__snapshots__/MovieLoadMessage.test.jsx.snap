// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when MovieLoadMessage then snapshot created 1`] = `
Array [
  <div
    className="modal-window-overlay-message"
  />,
  <div
    className="modal-window"
  >
    <div
      className="modal-window-loader"
    >
      Loading... Please wait!
    </div>
  </div>,
]
`;
